libs/common/src/helpers/payload-verification/is-verified-payload-by-type.ts

```diff
@@ -5,27 +5,43 @@ import { IsVerifiedPayloadByTypeParams } from '../../types/payload-verification'
  */
 export const isVerifiedPayloadByType = ({ payload, template }: IsVerifiedPayloadByTypeParams) => {
```

Comment:
This function was reworked as part of this ticket. To be honest, this function is depreciated and is not best practice due to a lot of flaws.

I've kept this rework and improved it as well to feature early returns etc, but we should really move away from using this due to how lax its validation is.

Reply:
I've optimised this as we were looping through lists unneccessarily several times, even if we were looking to get better error messages. I'm not sure the error messages will be too useful as they are only logs but I've kept them in an optimised and more useful manner